pool/share.go

```diff
@@ -96,7 +96,7 @@ func ppsEligibleShares(db *bolt.DB, max []byte) ([]*Share, error) {
 				return dbError(ErrDecode, desc)
 			}
 
-			if bytes.Compare(createdOnB, max) <= 0 {
+			if bytes.Compare(createdOnB, nanoToBigEndianBytes(max)) <= 0 {
```

Comment:
this is going to call `nanoToBigEndianBytes` on `max` for every single comparison. better to do this conversion outside the loop so this is done only once.

Reply:
nice find, fixed